test/TestUtil.js

```diff
@@ -85,7 +89,14 @@ exports.assertTrueEventually = function (taskAsyncFn, intervalMs = 100, timeoutM
 
         const timeoutTimer = setTimeout(() => {
             clearInterval(intervalTimer);
-            reject(new Error('Rejected due to timeout of ' + timeoutMs + 'ms. Errors occurred in order: \n\n' + errorString));
+
+            let errorString = '';
+            for (const error in errorsToCount) {
+                errorString += `\tThe following error happened ${errorsToCount[error]} times:\n\n\t${error} \n\n`;
+            }
+
+            reject(new Error('Rejected due to timeout of ' + timeoutMs
+                + 'ms. The following are the errors occurred and their counts: \n\n' + errorString));
```

Comment:
```suggestion
                + 'ms. The following are the errors that occurred and their counts: \n\n' + errorString));
```

Grammarly suggests this